<?xml version="1.0"?>

<launch>
    <arg name="input/pointcloud" default="/sensing/lidar/top/rectified/pointcloud"/>
    <arg name="output/objects" default="objects"/>
    <arg name="model_path" default="$(find-pkg-share lidar_centerpoint)/data"/>

    <node pkg="lidar_centerpoint" exec="lidar_centerpoint_node"
          name="lidar_centerpoint" output="screen">
        <remap from="~/input/pointcloud" to="$(var input/pointcloud)"/>
        <remap from="~/output/objects" to="$(var output/objects)"/>
        <param name="densification_base_frame" value="map"/>
        <param name="densification_past_frames" value="1"/>
        <param name="use_head_trt" value="true"/>
        <param name="trt_precision" value="fp16"/>
        <param name="vfe_pt_path" value="$(var model_path)/centerpoint_vfe.pt"/>
        <param name="head_onnx_path" value="$(var model_path)/centerpoint_head.onnx"/>
        <param name="head_engine_path" value="$(var model_path)/centerpoint_head.engine"/>
    </node>
</launch>
